modules/custom-status/custom-status.php

```diff
@@ -284,14 +287,46 @@ public function load_scripts_for_block_editor() {
 
 		$publish_guard_enabled = ( 'on' === VIP_Workflow::instance()->settings->module->options->publish_guard ) ? true : false;
 
+		$custom_statuses = $this->get_custom_statuses_with_editorial_metadata();
+
 		wp_localize_script( 'vip-workflow-block-custom-status-script', 'VW_CUSTOM_STATUSES', [
 			'current_user_id'          => get_current_user_id(),
 			'is_publish_guard_enabled' => $publish_guard_enabled,
-			'status_terms'             => $this->get_custom_statuses(),
+			'status_terms'             => $custom_statuses,
 			'supported_post_types'     => VIP_Workflow::instance()->get_supported_post_types(),
 		] );
 	}
 
+	/**
+	 * Get the custom statuses, with the editorial metadatas populated under meta.
+	 *
+	 * @return array $custom_statuses The custom statuses with editorial metadatas
+	 */
+	private function get_custom_statuses_with_editorial_metadata() {
```

Comment:
Any thought about making this default `get_custom_statuses()` behavior? One thing that trips me up is we set `meta[ self::METADATA_REQ_EDITORIAL_IDS_KEY ]` by default but `meta[ self::METADATA_REQ_EDITORIALS_KEY ]` is computed only when this function is called. I don't like that we need to remember which of these are provided by the base function and which are computed by other functions that need to be called later.

I think these keys ideally represent all available data on custom statuses, and have helper functions to provide computed results (e.g. `get_required_editorial_metadata($term_id)`) and not store the result in `meta`, since it's derived data. It's just handling both actual metadata and computed results in the same place that's confusing, I think.

Reply:
So, that is something I want to do but we need to finish the custom status migration before we can do that due to the init conflicts that arise. The priorities hacks worked sometimes, but they also caused issues sometimes. The problem is that EM's taxonomy gets used in CS and vice versa. 

I did some experimentation with #59 and it worked a lot better. So I've been waiting to get that in and then I'll tweak this.